pkg/liquidity-mining/test/BalancerMinter.test.ts

```diff
@@ -1,25 +1,249 @@
+        });
+
+        it('returns nothing', async () => {
+          expect(await mintStaticFunction()).to.be.deep.eq([]);
+        });
+      });
+    });
+  });
+
+  describe('mintMany / mintManyFor', () => {
+    let mintManyFunction: () => Promise<ContractReceipt>;
+    let mintManyStaticFunction: () => Promise<BigNumber>;
+    let gauges: string[];
+
+    const expectedMintManyForReturn = 83;
+
+    function itCallsVirtualHooks() {
+      it('calls internal _mintForMany virtual hook', async () => {
+        const receipt = await mintManyFunction();
+        expectEvent.inReceipt(receipt, 'MintForMany', { gauges, user: user.address });
+      });
+
+      it('returns the same value as the internal _mintForMany virtual hook', async () => {
+        expect(await mintManyStaticFunction()).to.be.eq(expectedMintManyForReturn);
+      });
+    }
+
+    sharedBeforeEach(async () => {
+      gauges = await Promise.all(range(5).map(randomAddress));
+      await minterContract.setMockMintForMany(expectedMintManyForReturn);
+    });
+
+    describe('mintMany', () => {
+      sharedBeforeEach(async () => {
+        mintManyFunction = async () => await (await minterContract.connect(user).mintMany(gauges)).wait();
+        mintManyStaticFunction = async () => await minterContract.connect(user).callStatic.mintMany(gauges);
+      });
+
+      itCallsVirtualHooks();
+    });
+
+    describe('mintManyFor', () => {
+      sharedBeforeEach(async () => {
+        mintManyFunction = async () =>
+          await (await minterContract.connect(minter).mintManyFor(gauges, user.address)).wait();
+        mintManyStaticFunction = async () =>
+          await minterContract.connect(minter).callStatic.mintManyFor(gauges, user.address);
+      });
+
+      context('when the caller is not allowed', () => {
+        it('reverts', async () => {
+          await expect(mintManyFunction()).to.be.revertedWith('Caller not allowed to mint for user');
+        });
+      });
+
+      context('when caller is allowed', () => {
+        sharedBeforeEach(async () => {
+          await minterContract.connect(user).setMinterApproval(minter.address, true);
+        });
+
+        itCallsVirtualHooks();
+      });
+    });
+
+    describe('mint_many', () => {
+      let gauges8: string[];
+
+      sharedBeforeEach(async () => {
+        mintManyFunction = async () => await (await minterContract.connect(user).mint_many(gauges8)).wait();
+        mintManyStaticFunction = async () => await minterContract.connect(user).callStatic.mint_many(gauges8);
+        gauges8 = await Promise.all(range(8).map(randomAddress)); // mint_many accepts a fixed length array of 8.
+      });
+
+      it('calls internal _mintFor virtual hook for each gauge', async () => {
+        const receipt = await mintManyFunction();
+        for (const gauge of gauges8) {
+          expectEvent.inReceipt(receipt, 'MintFor', { gauge, user: user.address });
+        }
+      });
```

Comment:
Why are you not calling `itCallsVirtualHooks();` here?

Reply:
because in this case `mint_many` ends up calling `_mintFor` repeateadly instead of `_mintForMany`, so it's not equivalent to `mintMany` :(